lib/rspec/matchers/built_in/match.rb

```diff
@@ -5,6 +5,22 @@ module BuiltIn
       # Provides the implementation for `match`.
       # Not intended to be instantiated directly.
       class Match < BaseMatcher
+        # Used to specify the captures we match against
+        # Accepts variadic non-named captures
+        # @return [self]
+        def with_captures(*captures)
+          @expected_captures = captures
+          extend WithCaptures
+        end
```

Comment:
Suggested error to raise for that condition? Any prior art in rspec for mis-use of a matcher?

Since we can accept a string either as expected or actual, we won't know at matcher-instantiation-time if it's a legal method call. So the earliest moment that we will know it's invalid is invocation of `.match(expected, actual)`, at which point we can raise an exception if both expected and actual are strings.